src/tools/fep.py

```diff
@@ -1,14 +1,34 @@
-"""module to work FEP calculations"""
+"""Module to help work on FEP calculations. This module provides the FEP class, that allows you to run FEP calculations on Deep Origin."""
 
 import importlib.resources
 import json
+import os
 from dataclasses import dataclass, fields
 from pathlib import Path
 from typing import Optional, Union
 
+import pandas as pd
 from beartype import beartype
 from box import Box
 from deeporigin import chemistry
+from deeporigin.data_hub import api
+from deeporigin.exceptions import DeepOriginException
+from deeporigin.tools.toolkit import _ensure_columns, _ensure_database
+
+# constants
+DB_ABFE = "ABFE2"
+DB_PROTEINS = "Proteins"
+DB_LIGANDS = "Ligands"
+COL_PROTEIN = "Protein"
+COL_LIGAND = "Ligand"
+COL_STEP = "Step"
+COL_JOBID = "JobID"
+COL_OUTPUT = "OutputFile"
+COL_RESULT = "ResultFile"
+COL_DELTA_G = "FEP ΔG (kcal/mol)"
+
```

Comment:
these constants identify the database that must exist. if it doesn't, it's created on the fly